libs/src/oracle-sdk-v2/services/oraclev1/ethers/factory.ts

```diff
@@ -147,11 +176,45 @@ export const Factory = (config: Config): [ServiceFactory, Api] => {
   const service = (handlers: Handlers): Service => {
     if (handlers.requests) events.on("requests", handlers.requests);
   };
+  async function queryRange(startBlock: number, endBlock: number) {
+    let rangeState = rangeStart({ startBlock, endBlock });
+    const { currentStart, currentEnd } = rangeState;
+    try {
+      await oo.update(currentStart, currentEnd);
+      rangeState = rangeSuccessDescending({ ...rangeState, multiplier: 1 });
+    } catch (err) {
+      rangeState = rangeFailureDescending(rangeState);
+    }
+  }
+  function queryLatestRequests(blocksAgo: number) {
```

Comment:
this is a new api call to each service which will cache events and emit any requests it finds to the oracle context